### DISCUSSION TITLE
Generating structural variants for SEQR

### DISCUSSION THREAD

**Discussion by illusional** [broadinstitute/seqr#2153] [General]
Hi!

We're interested in exploring structural variants in _seqr_, but interested to know if the pipeline to generate SVs that _seqr_ is using is publicly available. I can see an example of a multi-sample VCF that's expected by [the loading pipeline](https://github.com/populationgenomics/hail-elasticsearch-pipelines/blob/4766b07a63592931c3fb10945e45aa89643e48b7/sv_pipeline/genome/load_data_tests.py#L103-L113). We're aware this likely isn't a well supported pathway at the moment, and happy to contribute back anyway we can!

Also wondered if GATK-SV would be compatible, or if that's on the plan for SVs generated by the Broad.

FYI @lgruen / @vladsaveliev

**Comment by hanars** [comment]
I am not sure what the open source status of the code for generating the VCFs for SVs is at the moment - this data is generated by a different group we collaborate with and I'm not very familiar with the process they use. I do believe it is built off of GATK-SV so I think you could try to see if a "normal" GATK-SV VCF would work

**Comment by melnel000** [comment]
Dear seqr team

Our group is currently using the platform following the seqr workshop (in collaboration with ClinGEN/GREGoR) and finding it so useful! Thanks!

However, it's a real limitation to our case analysis abilities to only be able to screen short variants- we really need a solution which can address short variants as well as SVs and also repeat expansions. Are there any plans to offer users upload of a joint called SV callset via Terra (i.e. GATK-SV VCF)? Or will that remain a feature exclusively available to BROAD collaborators? If yes, is there any way to collaborate with seqr to access the SV functionality if you already have your own WGS data?

Kind regards,
Melissa

**Reply by hanars** [reply]
Hi Melissa,

We understand your request and we very much see the value in being able to analyze SVs along with short variants. We are planning to add this support for AnVIL users sometime in the next year. In the meantime, we can take occasional requests to add SV data to existing seqr projects. To do this, make sure you have a GATK-SV joint called VCF available in the "files" section of the workspace (the same place where your main VCF is stored). Then send us an email to seqr@broadinstitute.org with the name of the file and project you want us to load. These requests may take a couple weeks to complete, as we are not set up to load SVs in the way we are to load regular VCFs. We hope that this will be an adequate solution for you until we can add better support in the future.

Best,
Hana

**Comment by melnel000** [comment]
Dear Hana

I am going to apply for a Google Cloud research grant for cloud compute credits to run GATK-SV in cohort mode in Terra so that I can generate a joint called GATK-SV VCF for my cohort (which is already loaded in seqr).

In the meantime, I am paying to run a single sample using the GATK-SV single sample workflow. Would it be possible to request to load this SV data into seqr so that we could also look at SVs for this particular case? I have some internal avenues that I want to pursue for funding and it would be really great if I could demo a use case for the ability to view SVs alongside short variants for a sample in seqr.

I understand your workflow requires a GATK-SV joint called VCF but it’s going to take me some time to secure the funds to do this.

Kind regards,
Melissa

From: hanars ***@***.***>
Date: Tuesday, May 17, 2022 at 6:12 PM
To: broadinstitute/seqr ***@***.***>
Cc: melnel000 ***@***.***>, Comment ***@***.***>
Subject: Re: [broadinstitute/seqr] Generating structural variants for SEQR (Discussion #2153)

Hi Melissa,

We understand your request and we very much see the value in being able to analyze SVs along with short variants. We are planning to add this support for AnVIL users sometime in the next year. In the meantime, we can take occasional requests to add SV data to existing seqr projects. To do this, make sure you have a GATK-SV joint called VCF available in the "files" section of the workspace (the same place where your main VCF is stored). Then send us an email to ***@***.******@***.***> with the name of the file and project you want us to load. These requests may take a couple weeks to complete, as we are not set up to load SVs in the way we are to load regular VCFs. We hope that this will be an adequate solution for you until we can add better support in the future.

Best,
Hana

**Reply by hanars** [reply]
Hi Melissa

seqr can not accept single sample VCFs, only joint called VCFs, this is unfortunately non-negotiable so we can't load this data for you

Best
Hana

**Comment by AlessandroLas** [comment]
Is this true also for GATK-CNVs obtained by comparing a sample vs a group of normals?

**Reply by hanars** [reply]
Can you please be a bit clearer in your question? I don't know what you mean by "this"

**Reply by AlessandroLas** [reply]
Sorry.

I was referring to the discussion from above. 
You indicated that (for SVs) seqr can not accept single sample VCFs, only joint called VCFs.
Can seqr accept single sample VCFs of GATK-called CNVs?

**Reply by hanars** [reply]
No. seqr only ever accepts joint called data, regardless of data type

**Reply by AlessandroLas** [reply]
Thank you.

**Comment by paolomaccallini-hub** [comment]
Any news about the workflow to use for generating CNVs that can be analyzed in Seqr? Thanks!

**Reply by hanars** [reply]
Sorry, there is no update on this. The seqr team does not have any expertise on variant calling, we are dependent on the methods developed by other groups, and unfortunately that methods development has been going slower than we would have hoped

**Comment by Tulasi08** [comment]
Hi @hanars ,

Is there any update on supporting the upload of a joint called CNV/SV callset via AnVIL or for the local instance?

Thanks and Regards,
Tulasi S

**Reply by hanars** [reply]
No there is still no update. We will let you know once there is

**Comment by harmangea-chop** [comment]
@hanars does this mean that as of right now, CNVs are not yet supported by Seqr?

**Reply by hanars** [reply]
Correct
